src/views/app/Workforce-management/create-user.tsx

```diff
@@ -4,102 +4,157 @@ import ViewWrapper from '../../../components/layout/viewWrapper';
 import ButtonComponent from '../../../components/atoms/button';
 import { SelectComponent, SelectItemComponent } from '../../../components/atoms/select';
 import useModal from '../../../hooks/modal/useModal';
-import { ParamListBase, useIsFocused, useNavigation } from '@react-navigation/native';
+import { ParamListBase } from '@react-navigation/native';
 import useRole from '../../../hooks/role';
-import { useGetDepartmentsByCampusIdQuery, useGetDepartmentsQuery } from '../../../store/services/department';
-import { useCreateUserMutation, useGetUsersByDepartmentIdQuery } from '../../../store/services/account';
-import { IDepartment, IRegisterPayload } from '../../../store/types';
+import { useGetDepartmentsByCampusIdQuery } from '../../../store/services/department';
+import { useGetUsersByDepartmentIdQuery, useUploadUserMutation } from '../../../store/services/account';
+import { ICreateUserPayload, IDepartment, IRole } from '../../../store/types';
 import { Formik, FormikConfig } from 'formik';
-import { RegisterSchema } from '../../../utils/schemas';
+import { CreateUserSchema } from '../../../utils/schemas';
 import { Icon } from '@rneui/themed';
 import { THEME_CONFIG } from '../../../config/appConfig';
 import { NativeStackScreenProps } from '@react-navigation/native-stack';
 import { InputComponent } from '../../../components/atoms/input';
+import { useGetRolesQuery } from '../../../store/services/role';
 
-const createUser: React.FC<NativeStackScreenProps<ParamListBase>> = props => {
-    const { goBack, setOptions } = useNavigation();
+const CreateUser: React.FC<NativeStackScreenProps<ParamListBase>> = props => {
+    const { navigation } = props;
+    const { goBack } = navigation;
 
     const {
-        user: { campus },
+        user: { campus, userId },
     } = useRole();
 
-    const [icon, setIcon] = React.useState<{ name: string; type: string }>({
-        type: 'ionicon',
-        name: 'briefcase-outline',
-    });
-
     const [departmentId, setDepartmentId] = React.useState<IDepartment['_id']>(); //Just for 3P testing
+    const [roleId, setRoleId] = React.useState<IRole['_id']>();
 
     const { setModalState } = useModal();
 
     const {
         data: campusDepartments,
         refetch: refetchDepartments,
-        isSuccess: campusDepartmentsSuccess,
+        isFetching: isFetchingDepartments,
         isLoading: campusDepartmentsLoading,
     } = useGetDepartmentsByCampusIdQuery(campus._id);
 
+    const {
+        data: allRoles,
+        refetch: refetchRoles,
+        isFetching: isFetchingRoles,
+        isLoading: rolesIsLoading,
+    } = useGetRolesQuery();
+
     const {
         data: workers,
-        isLoading: workersLoading,
-        isSuccess: workerSuccess,
+        refetch: refetchWorkers,
+        isFetching: isFetchingWorkers,
     } = useGetUsersByDepartmentIdQuery(departmentId as string, {
         skip: !departmentId,
     });
 
```

Comment:
You might not need this. Check with @wealsegun to confirm backend validation for duplicate workers